utils/data.py

```diff
@@ -12,8 +12,10 @@ def __init__(self, *args, prefix=None, **kwargs):
     async def on_message(self, msg):
         if not self.is_ready() or msg.author.bot or not permissions.can_handle(msg, "send_messages"):
             return
-
-        await self.process_commands(msg)
+        if await listener.check_for_profanity(msg):
+            await msg.channel.send("Bad word detected!")
```

Comment:
Overall this does look pretty good. Maybe you could be a bit more specific on what you say when a bad word is detected. Like, you could have it say the name of the user that said the bad word, then say something about that language not being permissible on the server.

Reply:
Good thoughts, but detailed handling of bad words is out of scope for this task. It would be handled in #40. This task is to just demonstrate the ability to filter for a word (any arbitrary word or words) and do something (anything) in response.